mods/fire/init.lua

```diff
@@ -151,72 +161,92 @@ minetest.override_item("default:coalblock", {
 	end,
 })
 
--- Get sound area of position
-
-fire.D = 6 -- size of sound areas
 
-function fire.get_area_p0p1(pos)
-	local p0 = {
-		x = math.floor(pos.x / fire.D) * fire.D,
-		y = math.floor(pos.y / fire.D) * fire.D,
-		z = math.floor(pos.z / fire.D) * fire.D,
-	}
-	local p1 = {
-		x = p0.x + fire.D - 1,
-		y = p0.y + fire.D - 1,
-		z = p0.z + fire.D - 1
-	}
-	return p0, p1
+--
+-- Sounds
+--
+
+local handles = {}
+local timer = 0
+
+-- Parameters
+
+local radius = 6 -- Flame node search radius around player
+local cycle = 4 -- Cycle time for sound updates, in seconds
+				-- Maximum 4, limited by sample length
+
+-- Update sound for player
+
+local function fire.update_player_sound(player)
+	local player_name = player:get_player_name()
+	-- Search for flame nodes in radius around player
+	local ppos = player:getpos()
+	local fpos, num = minetest.find_nodes_in_area(
+		{x = ppos.x - radius, y = ppos.y - radius, z = ppos.z - radius},
+		{x = ppos.x + radius, y = ppos.y + radius, z = ppos.z + radius},
+		{"fire:basic_flame", "fire:permanent_flame"}
+	)
+	-- Total number of flames in radius
+	local flames = (num["fire:basic_flame"] or 0) +
+		(num["fire:permanent_flame"] or 0)
+	-- Stop previous sounds
+	if handles[player_name] then
+		minetest.sound_stop(handles[player_name])
+		handles[player_name] = nil
+	end	
+	-- If flames
+	if flames > 0 then
+		-- List midpoint is a position somewhat near the centre of all flames
+		local fposi = math.ceil(#fpos / 2)
+		-- Play sound
+		local handle = minetest.sound_play(
+			"fire_fire",
+			{
+				pos = fpos[fposi],
+				to_player = player_name,
+				gain = math.min(0.05 * math.sqrt(flames), 0.2),
+				max_hear_distance = 32,
+			}
+		)
+		-- Store sound handle for this player
+		if handle then
+			handles[player_name] = handle
```

Comment:
Even this is just one number in this table, the handle should be deleted when the player leaves.